ssz/sedes/list.py

```diff
@@ -163,6 +184,37 @@ def get_hash_tree_root(self, value: Iterable[TSerializable]) -> bytes:
 
         return mix_in_length(merkleize(merkle_leaves, limit=self.chunk_count()), len(value))
 
+    def get_hash_tree_root_and_leaves(self,
+                                      value: Iterable[TSerializable],
+                                      cache) -> bytes:
+        merkle_leaves = ()
+        if isinstance(self.element_sedes, BasicSedes):
+            serialized_items = tuple(
+                self.element_sedes.serialize(element)
+                for element in value
+            )
+            merkle_leaves = pack(serialized_items)
+        else:
+            has_get_hash_tree_root_and_leaves = hasattr(
+                self.element_sedes,
+                'get_hash_tree_root_and_leaves',
+            )
+            if has_get_hash_tree_root_and_leaves:
+                merkle_leaves = get_merkle_leaves_with_cache(
+                    value,
+                    self.element_sedes,
+                    cache,
+                )
+            else:
+                merkle_leaves = get_merkle_leaves_without_cache(value, self.element_sedes)
+
+        merkleize_result, cache = merkleize(
+            merkle_leaves,
+            limit=self.chunk_count(),
+            cache=cache,
+        )
+        return mix_in_length(merkleize_result, len(value)), cache
```

Comment:
I'm curious why these functions return the `cache`.

Reply:
Because I'm not certain we should implement `CacheObj` as a simple mutable `dict`, or we should design an `SSZCache` in the underlayer. 😬Right now it's always:
```python
def  merkleize_related_function_A(..., cache: CacheObj) -> Tuple[Hash32, CacheObj]:
    root, cache = merkleize_related_function_B(..., cache)
    ...

    return final_root, cache
```

A bit tedious.